esb: Improvement of the ESB driver to reduce fly time between emissio… (#9835)
nrfconnect/sdk-nrf: doc/nrf/protocols/esb/index.rst

The |NCS| provides the following example application that shows how to use the ESB protocol:

* :ref:`esb_prx_ptx`

.. _esb_fast_ramp_up:

Fast Ramp-up
============

The ESB driver is designed for bidirectional data between devices. 
This implies that the radio peripheral is continuously switching between reception and transmission mode.

> When switching to transmission mote, the radio peripheral needs some time to start-up internal analogs components.

Author: analog components is OK 👍

> On nRF51 and nRF24L series, a hardware-limited 130 µs delay is implemented.

Reviewer: ```suggestion
On the nRF51 and nRF24L Series devices, a hardware-limited 130 µs delay is implemented.
```

> If ESB connection is established only between nRF52 and/or nRF53 devices, this delay can be reduced to 40 µs.

Reviewer: ```suggestion
If the ESB connection is established only between nRF52 or nRF53 Series devices, this delay can be reduced to 40 µs.
```

Reviewer 2: Let's keep "end/or" to show that there are three possibilities: 52-52, 53-53, and 52-53.

Author: Done

> When ``use_fast_ramp_up`` parameter's value is true, the fast ramp-up will be enabled, resulting in reducing ramp-up delay to 40 µs.

Reviewer: ```suggestion
When the value of the ``use_fast_ramp_up`` parameter is ``true``, fast ramp-up is enabled, resulting in reduced ramp-up delay of 40 µs.
```

.. _esb_never_disable_tx:

Experimental feature: Never disable transmission stage
======================================================

> If reducing delay between TX FIFO filling and reception to minimum is needed, enabling fast ramp-up may not be sufficient.

Reviewer: ```suggestion
If you need to reduce the delay between TX FIFO filling and reception to minimum, enabling fast ramp-up might not be sufficient.
```

> In this case, you can use the :kconfig:option:`ESB_NEVER_DISABLE_TX` Kconfig option.

Reviewer 3: ```suggestion
In this case, you can use the :kconfig:option:`CONFIG_ESB_NEVER_DISABLE_TX` Kconfig option.
```

> It changes the driver's state machine so that if a packet is not acknowledged, the radio peripheral will remain in TXIDLE state instead of TXDISABLE when transmission is pending.

Reviewer 2: ```suggestion
This option changes the radio behavior so that if a packet is not acknowledged, the radio peripheral will remain in TXIDLE state instead of TXDISABLE when transmission is pending.
```

Author: 'The radio' refers to the radio peripheral (hardware module). 'the driver' refers to the code from nRF Connect library providing functions using the ESB protocol.
If you agree, I will only go with the first modification.

> This option changes the ESB driver's behavior so that if a packet is not acknowledged, the radio peripheral will remain in TXIDLE state instead of TXDISABLE when transmission is pending.

Reviewer: ```suggestion
It changes the ESB driver's behavior.
If a packet is not acknowledged, the radio peripheral remains in TXIDLE state instead of TXDISABLE when transmission is pending.
```

> This can reduce transmission delay below 100µs for a 32 bits payload but will consumes more energy as the radio transmitter stage will remained powered in-between transmissions.

Author: bits (4 bytes) => Clarification pending

> Using this experimental feature can reduce transmission delay below 100 µs for a 32 bits (4 bytes) payload, however, this process consumes more energy because the radio transmitter stage remains enabled when transmission is taking place.

Reviewer: ```suggestion
Using this experimental feature can reduce transmission delay below 100 µs for a 32 bits (four bytes) payload.
However, this process consumes more energy, because the radio transmitter stage remains enabled when transmission is taking place.
```
